### PR TITLE
global: major rewrite

### PR DESCRIPTION
**Pull request by JavierDelgadoFernandez**
* INCOMPATIBLE Brings a major rewrite.

Signed-off-by: Javier Delgado <javier.delgado.fernandez@cern.ch>

### CODE REVIEW

**Comment by JavierDelgadoFernandez** [comment]
Working on:
- [ ] Improving test coverage (Currently 90%).
- [ ] Fixing internationalization.
- [ ] Improving docstrings.

**Comment by JavierDelgadoFernandez** [comment]
@jirikuncar @lnielsen ping.

File: invenio_previewer/previewerext/csv_dthreejs.py [hunk 1]
```diff
@@ -1,79 +1,83 @@
...
         csvfile.seek(0)
         reader = csv.reader(csvfile, dialect)
         try:
             for row in reader:
-                u.feed(dialect.delimiter.join(row))
+                universal_detector.feed(
+                    dialect.delimiter.join(row).encode('utf-8'))
             is_valid = True
         except csv.Error as e:
             current_app.logger.debug(
```
Comment (JavierDelgadoFernandez) [inline invenio_previewer/previewerext/csv_dthreejs.py]: I was not able to create a test throwing this exception. The file is read line by line. Each line is split by the delimiter and it is not fed if the length of the result array does not match with the length of the first line. If you are using a "strange" delimiter, the code breaks in a previous line. It is the only line not tested and IMHO it should be removed.

File: .travis.yml [hunk 2]
```diff
@@ -1,80 +1,67 @@
 # -*- coding: utf-8 -*-
 #
-# This file is part of Invenio
-# Copyright (C) 2015 CERN
+# This file is part of Invenio.
+# Copyright (C) 2016 CERN.
```
Comment (jirikuncar) [inline .travis.yml]: 2015, 2016 ...

File: AUTHORS.rst [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ..
     This file is part of Invenio.
-    Copyright (C) 2015 CERN.
+    Copyright (C) 2016 CERN.
```
Comment (jirikuncar) [inline AUTHORS.rst]: 2015, 2016

File: CHANGES.rst [hunk 4]
```diff
@@ -21,9 +21,10 @@
     waive the privileges and immunities granted to it by virtue of its status
     as an Intergovernmental Organization or submit itself to any jurisdiction.
 
+
 Changes
 =======
 
-Version 0.1.0 (release 2015-08-13)
+Version 0.1.0 (released TBD)
```
Comment (jirikuncar) [inline CHANGES.rst]: keep the original date

File: README.rst [hunk 5]
```diff
@@ -1,6 +1,6 @@
 ..
     This file is part of Invenio.
-    Copyright (C) 2015 CERN.
+    Copyright (C) 2016 CERN.
```
Comment (jirikuncar) [inline README.rst]: 2015, 2016

File: RELEASE-NOTES.rst [hunk 6]
```diff
@@ -2,14 +2,14 @@
  Invenio-Previewer v0.1.0
 ==========================
 
-Invenio-Previewer v0.1.0 was released on August 13, 2015.
+Invenio-Previewer v0.1.0 was released on TBD, 2016.
```
Comment (jirikuncar) [inline RELEASE-NOTES.rst]: keep the original date or change all 0.1.0 to 1.0.0

File: babel.ini [hunk 7]
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 #
 # This file is part of Invenio.
-# Copyright (C) 2015 CERN.
+# Copyright (C) 2016 CERN.
```
Comment (jirikuncar) [inline babel.ini]: 2015

File: docs/conf.py [hunk 8]
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 #
 # This file is part of Invenio.
-# Copyright (C) 2015 CERN.
+# Copyright (C) 2016 CERN.
```
Comment (jirikuncar) [inline docs/conf.py]: 2015, 2016

File: docs/index.rst [hunk 9]
```diff
@@ -1,6 +1,6 @@
 ..
     This file is part of Invenio.
-    Copyright (C) 2015 CERN.
+    Copyright (C) 2016 CERN.
```
Comment (jirikuncar) [inline docs/index.rst]: 2015, 2016

File: examples/app.py [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+        'SQLALCHEMY_DATABASE_URI', 'sqlite:///test.db'),
+)
+Babel(app)
+FlaskCLI(app)
+InvenioDB(app)
+InvenioRecords(app)
+InvenioPreviewer(app)
+InvenioAssets(app)
+
+if __name__ == "__main__":
```
Comment (jirikuncar) [inline examples/app.py]: you can remove these 2 lines

File: invenio_previewer/bundles.py [hunk 11]
```diff
@@ -1,70 +1,101 @@
 # -*- coding: utf-8 -*-
 #
 # This file is part of Invenio.
-# Copyright (C) 2014, 2015 CERN.
+# Copyright (C) 2016 CERN.
```
Comment (jirikuncar) [inline invenio_previewer/bundles.py]: 2015, 2016

File: invenio_previewer/ext.py [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+        app.register_blueprint(blueprint)
+        app.extensions['invenio-previewer'] = self
+
+    def init_config(self, app):
+        """Initialize configuration."""
+        app.config.setdefault(
+            "PREVIEWER_BASE_TEMPLATE",
+            app.config.get("BASE_TEMPLATE",
+                           "invenio_previewer/base.html"))
+        app.config.setdefault(
```
Comment (jirikuncar) [inline invenio_previewer/ext.py]: please remove `SQLALCHEMY_TRACK_MODIFICATIONS` from this file

File: invenio_previewer/ext.py [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+    def init_app(self, app):
+        """Flask application initialization."""
+        self.init_config(app)
+        app.register_blueprint(blueprint)
+        app.extensions['invenio-previewer'] = self
+
+    def init_config(self, app):
+        """Initialize configuration."""
+        app.config.setdefault(
+            "PREVIEWER_BASE_TEMPLATE",
```
Comment (jirikuncar) [inline invenio_previewer/ext.py]: Please use single quotes `'` for strings and triple double quotes for docstrings `"""` everywhere. Thanks

File: invenio_previewer/previewerext/csv_dthreejs.py [hunk 14]
```diff
@@ -1,79 +1,83 @@
+# -*- coding: utf-8 -*-
+#
 # This file is part of Invenio.
-# Copyright (C) 2013, 2014, 2014, 2015 CERN.
+# Copyright (C) 2016 CERN.
```
Comment (jirikuncar) [inline invenio_previewer/previewerext/csv_dthreejs.py]: please keep old copyright years + 2016

File: invenio_previewer/previewerext/__init__.py [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Implementations of different viewers."""
+
+from . import mistune, csv_dthreejs, default, pdfjs, zip
```
Comment (jirikuncar) [inline invenio_previewer/previewerext/__init__.py]: please do not import these previewers here but use instead entry_points

File: invenio_previewer/previewerext/mistune.py [hunk 16]
```diff
@@ -1,45 +1,48 @@
+# -*- coding: utf-8 -*-
+#
 # This file is part of Invenio.
-# Copyright (C) 2013, 2014, 2015 CERN.
+# Copyright (C) 2016 CERN.
```
Comment (jirikuncar) [inline invenio_previewer/previewerext/mistune.py]: please keep passed copyright years

File: invenio_previewer/static/js/csv_previewer/loader.js [hunk 17]
```diff
@@ -18,12 +18,12 @@
  */
 'use strict';
 
-define(function(require) {
+define(function(requirebla) {
```
Comment (jirikuncar) [inline invenio_previewer/static/js/csv_previewer/loader.js]: `requirebla`?

File: invenio_previewer/templates/invenio_previewer/bottom.html [hunk 18]
```diff
@@ -1,7 +1,7 @@
-# -*- coding: utf-8 -*-
+{# -*- coding: utf-8 -*-
 #
 # This file is part of Invenio.
-# Copyright (C) 2015 CERN.
+# Copyright (C) 2016 CERN.
```
Comment (jirikuncar) [inline invenio_previewer/templates/invenio_previewer/bottom.html]: 2015, 2016

File: invenio_previewer/templates/invenio_previewer/bottom.html [hunk 19]
```diff
@@ -21,9 +21,6 @@
...
 # waive the privileges and immunities granted to it by virtue of its status
 # as an Intergovernmental Organization or submit itself to any jurisdiction.
+#}
 
-[tox]
-envlist = py27
-
-[testenv]
-commands = {envpython} setup.py test
+{%- extends "invenio_previewer/bottom_base.html" -%}
```
Comment (jirikuncar) [inline invenio_previewer/templates/invenio_previewer/bottom.html]: please configure you editor to end file with `\n\0`

File: invenio_previewer/templates/invenio_previewer/bottom_base.html [hunk 20]
```diff
@@ -1,23 +1,28 @@
-{#
+{# -*- coding: utf-8 -*-
+#
 # This file is part of Invenio.
-# Copyright (C) 2013, 2015 CERN.
+# Copyright (C) 2016 CERN.
```
Comment (jirikuncar) [inline invenio_previewer/templates/invenio_previewer/bottom_base.html]: keep old copyright years

File: invenio_previewer/templates/invenio_previewer/bottom_base.html [hunk 21]
```diff
@@ -80,19 +80,14 @@ <h3 id="embedModalLabel">Code to Embed</h3>
...
 # based on embed flag #}
 <div id="bottom">
-<div class="container row">
-  <div class="span5 offset1"><h3>Previewing: {{ f.name }}{{ f.superformat }}</h3></div>
-  <div class="span3">
-    {% if embed %}
-      <a role="button" class="btn btn-inverse" onclick="parent.window.location.href='{{ config.CFG_SITE_URL }}/record/{{ recid }}/'">View&nbsp;&nbsp<i class="icon-eye-open icon-white"></i></a>
-    {% else %}
+  <div class="container row">
+    <div class="span5 offset1"><h3>Previewing: {{ f.get_filename() }}</h3></div>
```
Comment (jirikuncar) [inline invenio_previewer/templates/invenio_previewer/bottom_base.html]: please make all strings translatable with `{{ _('Previewing: %(filename)s', filename=f.get_filename()) }}`

**Comment by jirikuncar** [comment]
- [ ] do not use `_base` templates
- [ ] configurable parent template

File: .travis.yml (hunk 2 above)
Comment (tiborsimko) [inline .travis.yml]: @JavierDelgadoFernandez Note that good code editors can amend the copyright years for you:
https://github.com/inveniosoftware/invenio/issues/172#issuecomment-42175134

**Comment by JavierDelgadoFernandez** [comment]
@lnielsen PR updated. Can you review it please?

**Pull request merged**